crates/ty_python_semantic/src/types/tuple.rs

```diff
@@ -213,29 +205,40 @@ impl<'db> TupleType<'db> {
         variable: Type<'db>,
         suffix: impl IntoIterator<Item = Type<'db>>,
     ) -> Option<Self> {
-        TupleType::new(db, VariableLengthTuple::mixed(prefix, variable, suffix))
+        TupleType::new(
+            db,
+            TupleSpec::new(db, VariableLengthTuple::mixed(prefix, variable, suffix)),
+        )
     }
 
     pub(crate) fn homogeneous(db: &'db dyn Db, element: Type<'db>) -> Option<Self> {
-        TupleType::new(db, TupleSpec::homogeneous(element))
+        TupleType::new(db, TupleSpec::homogeneous(db, element))
     }
 
-    // N.B. If this method is not Salsa-tracked, we take 10 minutes to check
-    // `static-frame` as part of a mypy_primer run! This is because it's called
-    // from `NominalInstanceType::class()`, which is a very hot method.
-    #[salsa::tracked(heap_size=ruff_memory_usage::heap_size)]
     pub(crate) fn to_class_type(self, db: &'db dyn Db) -> ClassType<'db> {
-        let tuple_class = KnownClass::Tuple
-            .try_to_class_literal(db)
-            .expect("Typeshed should always have a `tuple` class in `builtins.pyi`");
-
-        tuple_class.apply_specialization(db, |generic_context| {
-            if generic_context.variables(db).len() == 1 {
-                generic_context.specialize_tuple(db, self)
-            } else {
-                generic_context.default_specialization(db, Some(KnownClass::Tuple))
-            }
-        })
+        // N.B. If this inner function is not Salsa-tracked, we take 10 minutes to check
+        // `static-frame` as part of a mypy_primer run! This is because it's called
+        // from `NominalInstanceType::class()`, which is a very hot method.
+        #[salsa::tracked(heap_size=ruff_memory_usage::heap_size)]
+        fn to_class_type_inner<'db>(
+            db: &'db dyn Db,
+            tuple: TupleType<'db>,
+            _: (),
+        ) -> ClassType<'db> {
```

Comment:
Can't we implement `to_class_type` on the `TupleSpec` instead to avoid needing to intern `TupleType`? 

Or leave the function here but change the signoature to take a `TupleSpec` instead